ddtrace/_logger.py

```diff
@@ -37,7 +37,9 @@ def configure_ddtrace_logger():
 
     """
     ddtrace_logger = logging.getLogger("ddtrace")
-    ddtrace_logger.addHandler(logging.StreamHandler())
+    # don't add a handler if already configured by user
+    if not ddtrace_logger.handlers:
```

Comment:
I wonder if we should do something more like:

```python
stream_handler = logging.StreamHandler()

def configure_ddtrace_logger():
    if stream_handler not in ddtrace_logger:
        ddtrace_logger.addHandler(stream_handler)
```


not 100% sure what is the "best" approach, but this was a consideration that came up when reviewing.

Reply:
but what if the user has already added a FileHandler or some other kind of handler?

Reply:
the different is... "is our specific StreamHandler already set" vs "is any StreamHandler set". we'd want the former.

e.g.

```python
>>> import logging
>>> s1 = logging.StreamHandler()
>>> s2 = logging.StreamHandler()
>>> s3 = logging.StreamHandler()
>>> handlers = [s1, s2]
>>> s1 in handlers
True
>>> s2 in handlers
True
>>> s3 in handlers
False
```

Reply:
but again... is that what we want? I am not sure what normal/best practices are, but seems like a possible option.

Reply:
Shouldn't we only care about just adding a handler if the customer hasn't added one so we make sure we log either way, but also make sure that if the customer configures ddtrace logging, we don't mess with it?

Reply:
> Shouldn't we only care about just adding a handler if the customer hasn't added one

I think I was thinking about it more like "do not configure/add our handlers more than once", but not sure I know if it should be "do not configure if any handlers have been added".

🤔

We easily can end up in a case where we configure both the StreamHandler and a RotatingFileHandler, since our configuration can do both.

ok, not sure I have a good sense for the right answer here.

Reply:
Will sync with team on thoughts.